tests/testthat/test_util.R

```diff
@@ -837,6 +837,18 @@ test_that("get_mode", {
   expect_equal(as.character(res), "2019-01-02")
 })
 
+test_that("get_unknown_category_rows_index", {
+  train_df <- data.frame(x=c('a','b','c'),
+                         y=c('a','b','c'))
+  test_df <- data.frame(x=c('a','c','d'),
+                        y=c('b','e','c'))
+  unknown_vector <- get_unknown_category_rows_index_vector(test_df, train_df)
+  unknown_index <- get_row_numbers_from_index_vector(unknown_vector)
+  expect_equal(unknown_index,c(2,3))
+  restored <- restore_na(c('a','b','c'), c(2,4))
+  expect_equal(restored ,c('a',NA,'b',NA,'c'))
```

Comment:
Can you add a test case that covers the output that has NAs in both ends just in case?

```
c(NA, 'a',NA,'b',NA,'c', NA)
```